pipes/mp_proposal_pipe.py

```diff
@@ -0,0 +1,148 @@
+
+# Paths
+mp_proposal_tsv_path = os.path.join("data", "raw", "vaski", "LegislativeMotion_fi.tsv")
+mp_proposals_csv = os.path.join("data", "preprocessed", "mp_proposals.csv")
+mp_proposal_signatures_csv = os.path.join("data", "preprocessed", "mp_proposal_signatures.csv")
+handling_tsv_path = os.path.join("data", "raw", "vaski", "KasittelytiedotValtiopaivaasia_fi.tsv")
+
+# Namespaces
+NS = {
+    'asi': 'http://www.vn.fi/skeemat/asiakirjakooste/2010/04/27',
+    'asi1': 'http://www.vn.fi/skeemat/asiakirjaelementit/2010/04/27',
+    'met': 'http://www.vn.fi/skeemat/metatietokooste/2010/04/27',
+    'met1': 'http://www.vn.fi/skeemat/metatietoelementit/2010/04/27',
+    'org': 'http://www.vn.fi/skeemat/organisaatiokooste/2010/02/15',
+    'org1': 'http://www.vn.fi/skeemat/organisaatioelementit/2010/02/15',
+    'sis': 'http://www.vn.fi/skeemat/sisaltokooste/2010/04/27',
+    'sis1': 'http://www.vn.fi/skeemat/sisaltoelementit/2010/04/27',
+    'vml': 'http://www.eduskunta.fi/skeemat/mietinto/2011/01/04',
+    'vsk': 'http://www.eduskunta.fi/skeemat/vaskikooste/2011/01/04',
+    'vsk1': 'http://www.eduskunta.fi/skeemat/vaskielementit/2011/01/04',
+    'saa': 'http://www.vn.fi/skeemat/saadoskooste/2010/04/27',
+    'saa1': 'http://www.vn.fi/skeemat/saadoselementit/2010/04/27',
+    'vas': 'http://www.eduskunta.fi/skeemat/vastalause/2011/01/04',
+    'jme': 'http://www.eduskunta.fi/skeemat/julkaisusiirtokooste/2011/12/20',
+    'ns11': 'http://www.eduskunta.fi/skeemat/siirto/2011/09/07',
+    'ns4': 'http://www.eduskunta.fi/skeemat/siirtoelementit/2011/05/17',
+    's359': 'http://www.vn.fi/skeemat/metatietoelementit/2010/04/27',
+    's360': 'http://www.vn.fi/skeemat/metatietoelementit/2010/04/27',
+    'sii': 'http://www.eduskunta.fi/skeemat/siirtokooste/2011/05/17',
+    'sii1': 'http://www.eduskunta.fi/skeemat/siirtoelementit/2011/05/17',
+    'he': 'http://www.vn.fi/skeemat/he/2010/04/27',
+    'tau': 'http://www.vn.fi/skeemat/taulukkokooste/2010/04/27',
+    'mix': 'http://www.loc.gov/mix/v20',
+    'narc': 'http://www.narc.fi/sahke2/2010-09_vnk',
+    'xsi': 'http://www.w3.org/2001/XMLSchema-instance',
+    'def': 'http://www.eduskunta.fi/skeemat/siirtokooste/2011/05/17',
+    'eka': "http://www.eduskunta.fi/skeemat/eduskuntaaloite/2012/08/10"
+}
+
+def preprocess_data():
+    os.makedirs(os.path.dirname(mp_proposals_csv), exist_ok=True)
+    mpp_df = pd.read_csv(mp_proposal_tsv_path, sep="\t")
+    handling_df = pd.read_csv(handling_tsv_path, sep="\t")
+
+    mpp_records = []            # government_proposals rows
+    sgn_records = []  
+
+    conn = psycopg2.connect(
+        database="postgres",
+        host="db",
+        user="postgres",
+        password="postgres",
+        port="5432"
+        )
+    
+    cur = conn.cursor()          
+
+    for mpp_xml_str in mpp_df.get("XmlData", []):
```

Comment:
täl ei nyt meiän tapaukses oo välii mut monis tilanteis näin pitkäikäset tietokantayhteydet on vähän hasardeja, tätä nyt ajetaan tasan yks kerrallaan mut vaik jossai hyvin rinnakkaistetus serveris tällaset tosi helposti tukkii jytynki palvelimen aik turhaan, eli pitäs jäsennellä koodi niin et tietokantayhteys avataan ain nii harvoin ku mahollista ja niin pieneks aikaa ku mahollista (usein myös tässä järkässä)